
syntax = "proto3";
package comfy;


service Comfy {
  rpc SetImage(Image) returns (Empty) {}
  rpc SetMask(Image) returns (Empty) {}
  rpc SetOptions(Options) returns (Empty) {}
  rpc Inpaint(Empty) returns (Image) {}
  rpc Img2Img(Empty) returns (Image) {}
}

enum ImgType {
  NONE = 0;
  MONO = 1;
  DUO = 2;
  RGB = 3;
  RGBA = 4;
}

message Empty {}

message Position {
  int32 x_pos = 1;
  int32 y_pos = 2;
}

message PatchPixels {
  bytes pixels = 1;
}

message ImgInfo {
  int32 width = 1;
  int32 height = 2;
  ImgType img_type = 3;
}

message Image {
  ImgInfo info = 1;
  bytes pixels = 2;
}

message InpaintData {
  Image img = 1;
  Image mask = 2;
}

message Options {
  string prompt = 1;
}

